plugins/funder.c

```diff
@@ -752,64 +773,179 @@ json_funderupdate(struct command *cmd,
-		   p_opt("per_channel_max", param_sat, &per_channel_max),
-		   p_opt("reserve_tank", param_sat, &reserve_tank),
-		   p_opt("fuzz_percent", param_number, &fuzz_factor),
-		   p_opt("fund_probability", param_number, &fund_probability),
+		   p_opt_def("policy", param_funder_opt, &opt,
+			     current_policy->opt),
+		   p_opt_def("policy_mod", param_policy_mod, &mod,
+			     current_policy->mod),
+		   p_opt_def("leases_only", param_bool, &leases_only,
+			     current_policy->leases_only),
+		   p_opt_def("min_their_funding", param_sat,
+			     &min_their_funding,
+			     current_policy->min_their_funding),
+		   p_opt_def("max_their_funding", param_sat,
+			     &max_their_funding,
+			     current_policy->max_their_funding),
+		   p_opt_def("per_channel_min", param_sat,
+			     &per_channel_min,
+			     current_policy->per_channel_min),
+		   p_opt_def("per_channel_max", param_sat,
+			     &per_channel_max,
+			     current_policy->per_channel_max),
+		   p_opt_def("reserve_tank", param_sat, &reserve_tank,
+			     current_policy->reserve_tank),
+		   p_opt_def("fuzz_percent", param_number,
+			     &fuzz_factor,
+			     current_policy->fuzz_factor),
+		   p_opt_def("fund_probability", param_number,
+			     &fund_probability,
+			     current_policy->fund_probability),
+		   p_opt("leases_available", param_bool, &leases_avail),
+		   p_opt_def("channel_fee_base_max_msat", param_msat,
+			     &channel_base_max,
+			     current_policy->rates ?
+				current_policy->rates->channel_base_max :
+				AMOUNT_MSAT(-1)),
+		   p_opt("channel_fee_basis_max", param_number,
+			     &channel_basis_max),
+		   p_opt("lease_fee_base_msat", param_sat, &lease_fee_base),
+		   p_opt("lease_fee_basis", param_number,
+			  &lease_fee_basis),
 		   NULL))
 		return command_param_failed();
 
-	if (opt)
-		policy->opt = *opt;
-	if (mod)
-		policy->mod = *mod;
-	if (min_their_funding)
-		policy->min_their_funding = *min_their_funding;
-	if (max_their_funding)
-		policy->max_their_funding = *max_their_funding;
-	if (per_channel_min)
-		policy->per_channel_min = *per_channel_min;
-	if (per_channel_max)
-		policy->per_channel_max = *per_channel_max;
-	if (reserve_tank)
-		policy->reserve_tank = *reserve_tank;
-	if (fuzz_factor)
-		policy->fuzz_factor = *fuzz_factor;
-	if (fund_probability)
-		policy->fund_probability = *fund_probability;
+	policy->opt = *opt;
+	policy->mod = *mod;
+	policy->min_their_funding = *min_their_funding;
+	policy->max_their_funding = *max_their_funding;
+	policy->per_channel_min = *per_channel_min;
+	policy->per_channel_max = *per_channel_max;
+	policy->reserve_tank = *reserve_tank;
+	policy->fuzz_factor = *fuzz_factor;
+	policy->fund_probability = *fund_probability;
+	policy->leases_only = *leases_only;
+
+	/* No rates if we're turning them off */
+	if (leases_avail && !*leases_avail) {
+		new_rates = tal_free(new_rates);
+		plugin_log(cmd->plugin, LOG_DBG,
+			   "leases are now off");
+	}
```

Comment:
Hmm, this is awkward.  Normally we'd set defaults, and let them alter them.

For example, if you are offering all zero leases, we should consider that "don't advertize lease" right?  Maybe that's too awkward: we could do a single `lease=base/basis/weight/feebasemax/feeratemax`, and if you omit a number it's the default?  (Pick a sane default!)

Reply:
>  lease=base/basis/weight/feebasemax/feeratemax,

This is the sort of problem that YAML solves :wink: 

I don't have anything against defaults, but I did want users to explicitly consider their values (and know how to change them if they wanted to). Forcing them to be set upfront means you have to consider the entire picture before you can enable them -- you can update them singly after the fact but by providing all of them you have convinced me that you've put a little bit of thought into it (and that you know how to change them later.. maybe).

I need to work on the documentation for funder, which will include a sample configuration. This will be the source of the default config values then.